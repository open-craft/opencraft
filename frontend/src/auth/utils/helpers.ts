import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
