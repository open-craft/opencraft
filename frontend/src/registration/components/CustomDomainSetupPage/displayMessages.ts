const messages = {
  configureOwnDomain: {
    defaultMessage: 'Configure your domain',
    description: ''
  },
  configureOwnDomainInstructions: {
    defaultMessage:
      'You will have to update the settings on your domain to be able to ' +
      'continue with the registration process.',
    description: ''
  },
  changesMade: {
    defaultMessage: 'I have made these changes',
    description: ''
  },
  buyDomainStep: {
    defaultMessage:
      'If you already have a domain, skip this step. ' +
      "If you don't have a domain name yet, we recommend " +
      'you go to Gandi.net to order one. Come back to this page after ordering it.',
    description: ''
  },
  setDnsRecordsStep: {
    defaultMessage:
      'Go to your domainâ€™s DNS configuration settings and add the following entries:',
    description: ''
  },
  saveStep: {
    defaultMessage:
      'Save the changes and activate them if required. It might take a while for ' +
      'the domain changes to propagate (up to 6 hours).',
    description: ''
  }
};

export default messages;
