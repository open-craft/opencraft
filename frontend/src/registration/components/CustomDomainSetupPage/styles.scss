@import '~styles/theme';

.div-fill {
  width: 100%;
}

.custom-domain-setup {
  width: 100%;
  text-align: center;

  h1 {
    font-family: $playfair-font;
    font-size: 30px;
    font-weight: bold;
    letter-spacing: 1.44px;
    color: #1e4c59 !important;
    margin-bottom: 20px;
  }

  p {
    font-family: $chivo-font;
    font-size: 18px;
    letter-spacing: 0.54px;
    color: #0f1f24;
  }

  .instruction-box {
    background-color: $light-blue-3;
    margin: 20px 0;
    padding: 20px;

    .number {
      font-family: $playfair-font;
      font-size: 40px;
      font-weight: bold;
      letter-spacing: 0.4px;
      color: #0f1f24;
    }

    .instruction {
      text-align: left;
      font-family: $chivo-font;
      font-size: 18px;
      letter-spacing: 0.54px;
      color: #0f1f24;
    }

    .extra {
      table {
        width: 100%;
        border: 1px solid black;

        th {
          border: 1px solid black;
          background-color: #6d9cae;
          font-family: $chivo-font;
          font-weight: normal;
          font-size: 18px;
          letter-spacing: 0.54px;
          color: white;
          padding: 8px 0;
        }

        td {
          border: 1px solid black;
          font-family: $chivo-font;
          font-size: 14px;
          letter-spacing: 0.42px;
          color: #0f1f24;
          padding: 8px 0;
        }
      }
    }
  }
}
