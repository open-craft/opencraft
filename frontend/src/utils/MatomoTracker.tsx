import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
