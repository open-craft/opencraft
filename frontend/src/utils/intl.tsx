import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
