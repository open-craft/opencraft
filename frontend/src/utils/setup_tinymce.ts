import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
