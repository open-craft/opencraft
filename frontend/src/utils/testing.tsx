import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
