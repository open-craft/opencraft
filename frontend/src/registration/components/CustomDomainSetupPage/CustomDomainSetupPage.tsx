import { GANDI_REFERRAL_LINK, RegistrationSteps } from 'global/constants';
import { RootState } from 'global/state';
import * as React from 'react';
import { connect } from 'react-redux';
import { WrappedMessage } from 'utils/intl';
import { RedirectToCorrectStep } from 'registration/components';
import { RegistrationStateModel } from 'registration/models';
import { Button, Col, Row, Spinner } from 'react-bootstrap';
import { goToNextStep } from '../../actions';
import { RegistrationPage } from '../RegistrationPage';
import messages from './displayMessages';
import './styles.scss';

interface ActionProps {
  goToNextStep: Function;
}

interface State {
  externalDomain: string;
}

interface Props extends ActionProps {}
interface StateProps extends RegistrationStateModel {}

interface Props extends StateProps, ActionProps {}

interface StepBoxProps {
  step: number;
  instruction: any;
  extra?: any;
}
export const StepBox: React.SFC<StepBoxProps> = (props: StepBoxProps) => (
  <div className="instruction-box">
    <Row>
      <Col lg="1" className="number">
        {props.step}
      </Col>
      <Col lg="11" className="instruction">
        {props.instruction}
      </Col>
    </Row>
    {props.extra && (
      <Row>
        <Col className="extra">{props.extra}</Col>
      </Row>
    )}
  </div>
);

@connect<{}, ActionProps, {}, Props, RootState>(
  (state: RootState) => ({
    ...state.registration
  }),
  {
    goToNextStep
  }
)
export class CustomDomainSetupPage extends React.PureComponent<Props, State> {
  constructor(props: Props, state: State) {
    super(props);
  }

  public render() {
    const setDnsRecordsStepExtraContent = (
      <table>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Value</th>
        </tr>
        <tr>
          <td>{this.props.registrationData.externalDomain}</td>
          <td>CNAME</td>
          <td>haproxy.net.opencraft.hosting.</td>
        </tr>
        <tr>
          <td>
            *.
            {this.props.registrationData.externalDomain}
          </td>
          <td>CNAME</td>
          <td>haproxy.net.opencraft.hosting.</td>
        </tr>
      </table>
    );

    const buyDomainInstruction = (
      <WrappedMessage
        messages={messages}
        id="buyDomainStep"
        values={{
          a: (...chunks: string[]) => (
            <a
              href={GANDI_REFERRAL_LINK}
              target="_blank"
              rel="noopener noreferrer"
            >
              {chunks}
            </a>
          )
        }}
      />
    );

    return (
      <div className="div-fill">
        <RegistrationPage
          title="Pro & Teacher Account"
          subtitle="Create your own Open edX instance now."
          currentStep={1}
        >
          <RedirectToCorrectStep
            currentPageStep={1}
            currentRegistrationStep={this.props.currentRegistrationStep}
          />
          <div className="custom-domain-setup">
            <h1>
              <WrappedMessage messages={messages} id="configureOwnDomain" />
            </h1>

            <p>
              <WrappedMessage
                messages={messages}
                id="configureOwnDomainInstructions"
              />
            </p>

            <StepBox step={1} instruction={buyDomainInstruction} />
            <StepBox
              step={2}
              instruction={
                <WrappedMessage messages={messages} id="setDnsRecordsStep" />
              }
              extra={setDnsRecordsStepExtraContent}
            />
            <StepBox
              step={3}
              instruction={<WrappedMessage messages={messages} id="saveStep" />}
            />

            <Button
              className="pull-left loading"
              variant="primary"
              size="lg"
              disabled={this.props.loading}
              onClick={() => {
                this.props.goToNextStep(RegistrationSteps.INSTANCE);
              }}
            >
              {this.props.loading && (
                <Spinner animation="border" size="sm" className="spinner" />
              )}
              <WrappedMessage messages={messages} id="changesMade" />
            </Button>
          </div>
        </RegistrationPage>
      </div>
    );
  }
}
