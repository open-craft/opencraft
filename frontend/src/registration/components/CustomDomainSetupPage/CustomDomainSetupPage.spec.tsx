import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { CustomDomainSetupPage } from './CustomDomainSetupPage';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<CustomDomainSetupPage />).toJSON();
    expect(tree).toMatchSnapshot();
});
