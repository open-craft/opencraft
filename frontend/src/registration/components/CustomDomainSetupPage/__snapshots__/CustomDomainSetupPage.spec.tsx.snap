// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="div-fill"
>
  <div
    className="registration-page"
  >
    <h1>
      Pro & Teacher Account
    </h1>
    <h2>
      Create your own Open edX instance now.
    </h2>
    <div
      className="step-bar-container"
    >
      <div
        className="d-flex justify-content-center"
      >
        <div
          className="step-bar"
        >
          <svg
            className="circled-number"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              className="outer-circle"
              cx={10}
              cy={10}
              r={10}
            />
            <circle
              className="inner-circle"
              cx={10}
              cy={10}
              r={7}
              strokeWidth={2}
            />
            <text
              className="circle-text"
              color="#1e4c59"
              dy="0.4em"
              fontSize="10px"
              strokeWidth="1px"
              textAnchor="middle"
              x="50%"
              y="50%"
            >
              1
            </text>
          </svg>
          <span
            className="line"
          />
          <svg
            className="step"
            viewBox="0 0 10 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx={5}
              cy={5}
              fill="#6d9cae"
              r={5}
            />
          </svg>
          <span
            className="line"
          />
          <svg
            className="step"
            viewBox="0 0 10 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx={5}
              cy={5}
              fill="#6d9cae"
              r={5}
            />
          </svg>
          <span
            className="line"
          />
          <svg
            className="step"
            viewBox="0 0 10 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx={5}
              cy={5}
              fill="#6d9cae"
              r={5}
            />
          </svg>
          <span
            className="line"
          />
        </div>
      </div>
    </div>
    <div
      className="registration-page-container"
    >
      <div
        className="registration-page-content row"
      >
        <div
          className="custom-domain-setup"
        >
          <h1>
            Configure your domain
          </h1>
          <p>
            You will have to update the settings on your domain to be able to continue with the registration process.
          </p>
          <div
            className="instruction-box"
          >
            <div
              className="row"
            >
              <div
                className="number col-lg-1"
              >
                1
              </div>
              <div
                className="instruction col-lg-11"
              >
                If you already have a domain, skip this step. If you don't have a domain name yet, we recommend you go to Gandi.net to order one. Come back to this page after ordering it.
              </div>
            </div>
          </div>
          <div
            className="instruction-box"
          >
            <div
              className="row"
            >
              <div
                className="number col-lg-1"
              >
                2
              </div>
              <div
                className="instruction col-lg-11"
              >
                Go to your domainâ€™s DNS configuration settings and add the following entries:
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="extra col"
              >
                <table>
                  <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      Type
                    </th>
                    <th>
                      Value
                    </th>
                  </tr>
                  <tr>
                    <td>
                      
                    </td>
                    <td>
                      CNAME
                    </td>
                    <td>
                      haproxy.net.opencraft.hosting
                    </td>
                  </tr>
                  <tr>
                    <td>
                      *.
                      
                    </td>
                    <td>
                      CNAME
                    </td>
                    <td>
                      haproxy.net.opencraft.hosting
                    </td>
                  </tr>
                </table>
              </div>
            </div>
          </div>
          <div
            className="instruction-box"
          >
            <div
              className="row"
            >
              <div
                className="number col-lg-1"
              >
                3
              </div>
              <div
                className="instruction col-lg-11"
              >
                Save the changes and activate them if required. It might take a while for the domain changes to propagate (up to 6 hours).
              </div>
            </div>
          </div>
          <button
            className="pull-left loading btn btn-primary btn-lg"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            I have made these changes
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
